src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -120,12 +134,12 @@ Namespace Microsoft.CodeAnalysis.Semantics
                         Dim parameter = parameters(index)
                         Dim value = Create(argument)
 
-                        Return New Argument(
+                        Return New VisualBasicArgument(
                             kind,
                             parameter,
                             value,
-                            inConversion:=Nothing,
-                            outConversion:=Nothing,
+                            inConversion:=New Conversion(Conversions.Identity),
+                            outConversion:=New Conversion(Conversions.Identity),
                             semanticModel:=_semanticModel,
                             syntax:=value.Syntax,
                             type:=Nothing,
```

Comment:
It looks like the only difference between `If` and `Else` is `kind`. Consider calculating `kind` inside the `If` and move the remainder outside the `If`.